CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+cmake_minimum_required (VERSION 3.9)
+project(maxminddb
+  LANGUAGES C
+  VERSION 1.4.3
+)
+set(MAXMINDDB_SOVERSION 0.0.7)
+add_definitions(-DPACKAGE_VERSION="${PROJECT_VERSION}")
+
+include(CheckTypeSize)
+check_type_size("unsigned __int128" UINT128)
+check_type_size("unsigned int __attribute__((mode(TI)))" UINT128_USING_MODE)
+if(HAVE_UINT128)
+  set(MMDB_UINT128_USING_MODE 0)
+  set(MMDB_UINT128_IS_BYTE_ARRAY 0)
+elseif(HAVE_UINT128_USING_MODE)
+  set(MMDB_UINT128_USING_MODE 1)
+  set(MMDB_UINT128_IS_BYTE_ARRAY 0)
+else()
+  set(MMDB_UINT128_USING_MODE 0)
+  set(MMDB_UINT128_IS_BYTE_ARRAY 1)
+endif()
+
+set(MMDB_UINT128_USING_MODE, HAVE__int128)
```

Comment:
Why set it here as well as in the if/elseif/else?

Reply:
Thanks. This was leftover.